orchagent/neighorch.cpp

```diff
@@ -1133,8 +1133,30 @@ void NeighOrch::doVoqSystemNeighTask(Consumer &consumer)
             }
 
             if (m_syncdNeighbors.find(neighbor_entry) == m_syncdNeighbors.end() ||
-                    m_syncdNeighbors[neighbor_entry].mac != mac_address)
+                    m_syncdNeighbors[neighbor_entry].mac != mac_address ||
+                    m_syncdNeighbors[neighbor_entry].voq_encap_index != encap_index)
             {
+                // Handle encap index change. SAI does not support change of encap index for
+                // existing neighbors. Remove the neighbor but do not errase from consumer sync
+                // buffer. The next iteration will add the neighbor back with new encap index
```

Comment:
Deleing neigh and adding it back may churn routers resolving over that neigh. So routes get deleted and added back.  I guess encap index change is common. Probably should be seen in syncd restart, as you mentioned, or linecard reload? Ideally, encap index change should be handled by SAI just like how it handles MAC change today.

Reply:
typo: i meant to say encap index change is NOT common.

Reply:
I agree. SAI should handle the encap index change similar to mac change. But current SAI does not support it. Till SAI supports it, we will have to live with this delete and re-add approach. Once SAI is enhanced for the encap index change handling, we can change this.

Reply:
@vganesan-nokia can you create issue in SONiC for tracking SAI not supporting set operation

Reply:
> @vganesan-nokia can you create issue in SONiC for tracking SAI not supporting set operation

Yes. Created issue https://github.com/Azure/sonic-buildimage/issues/7879. Thanks

Reply:
i was discussing with @prsunny and we think better option will be call SAI api with set and if it fails (which it will) then we should do remove/add neigh. That way once the attribute is supported we don't need any sonic change going forward.

Reply:
There is a problem with this approach. In current SAI (w.r.t brcm SAI 5.0), when set is done for encap index attribute in the neighbor record, SAI's neighbor entry set attribute API returns SAI_STATUS_FAILURE. When there is not SUCCESS from SAI, syncd sends orchagent shut down.

Reply:
Fixed. Since syncd is not sending orchagent shutdown for SET failures anymore, changes are done as suggested.